Dalamud/Game/Network/Structures/MarketBoardCurrentOfferings.cs

```diff
@@ -223,3 +223,137 @@ internal ItemMateria()
         }
     }
 }
+
+/// <summary>
+/// An interface that represents the current market board offerings.
+/// </summary>
+public interface IMarketBoardCurrentOfferings
+{
+    /// <summary>
+    /// Gets the list of individual item listings.
+    /// </summary>
+    List<IMarketBoardItemListing> ItemListings { get; }
+
+    /// <summary>
+    /// Gets the listing end index.
+    /// </summary>
+    int ListingIndexEnd { get; }
+
+    /// <summary>
+    /// Gets the listing start index.
+    /// </summary>
+    int ListingIndexStart { get; }
+
+    /// <summary>
+    /// Gets the request ID.
+    /// </summary>
+    int RequestId { get; }
+}
+
+/// <summary>
+/// An interface that represents the current market board offering of a single item from the <see cref="IMarketBoardCurrentOfferings"/>.
+/// </summary>
+public interface IMarketBoardItemListing
+{
+    /// <summary>
+    /// Gets the artisan ID.
+    /// </summary>
+    ulong ArtisanId { get; }
```

Comment:
The interfaces mean that if we ever need to change the internal implementation we can guarantee we don't make a mistake that ends up being breaking. It also allows us to add fields that are not visible to the plugins but are internally.
I don't have an issue either way and if you think it makes more sense to leave them as uninterfaced I can do so

Reply:
This kinda ties into my commend/RFJ on the other PR. See, e.g. https://softwareengineering.stackexchange.com/q/390084 for a larger commentary on when to use/not use interfaces.

I'll think about this some more and do some more research, but my current thought is that if there's no compelling reason to provide an interface purely for data classes, we probably shouldn't.

Reply:
No strong opinion on the interface, but as for naming, the names came from the Sapphire field names at the time, and aren't necessarily the best names we can give these. Definitely in favor of using custom names and just using `JsonProperty` to send the API ones to Universalis.

Reply:
Here are my thoughts on the renames I'd do
ArtisanId -> CreatorId
CatalogId -> ItemId

Everything else looks clear enough to me

Reply:
What value does ArtisanId/CreatorId have? Is it something that could be removed like RetainerOwnerId?

Reply:
Probably, though it's typically not even updated (ContentIdNameReq) - it's only sent by the server when you hover an item, and the packet data is just 8 bytes, so it's a pain to detect on new patches.

Reply:
ArtisanId appears to be the CharacterId of the person who crafted the item(makes sense I suppose). 0 when it's not crafted by anyone. I'm trying to think of how someone would misuse it and I don't really have any use cases

Reply:
@karashiiro Do you know what CatalogId2 represents on MarketBoardHistory?
As far as I can see it either replicates CatalogId or is 0. We could potentially remove it from the interface so as to not confuse plugin devs

Reply:
> ArtisanId appears to be the CharacterId of the person who crafted the item(makes sense I suppose). 0 when it's not crafted by anyone. I'm trying to think of how someone would misuse it and I don't really have any use cases

I was more asking for the value in having it, what benefits does it provide? I doubt that data will matter to most people.

Reply:
I guess I'm on the side of having it in there as a plugin could come up with a use for it in the future. If it has the potential to cause problems sure remove it but otherwise no reason not to keep it in.

Have made the renames as requested and removed NextCatalogId(as it makes no sense in the context of the public interface)

Reply:
> Do you know what CatalogId2 represents on MarketBoardHistory?

As (it sounds like) you've figured out, it's really a `NextCatalogId` - it's useful for collecting the result set pages, but doesn't need to be exposed, since that pagination should be handled by `NetworkHandlers` already.